src/stream/src/executor/exchange/input.rs

```diff
@@ -358,3 +362,16 @@ pub(crate) fn new_input(
 
     Ok(input)
 }
+
+impl DispatcherMessageBatch {
```

Comment:
Can change to return with an iterator with [auto_enums](https://docs.rs/auto_enums/latest/auto_enums/) to avoid creating a vec for each message.

```
impl DispatcherMessageBatch {
    fn into_messages(self) -> impl Iterator<Item = DispatcherMessage> {
        #[auto_enum(Iterator)]
        match self {
            DispatcherMessageBatch::BarrierBatch(barriers) => barriers
                .into_iter()
            DispatcherMessageBatch::Chunk(c) => once(DispatcherMessage::Chunk(c)),
            DispatcherMessageBatch::Watermark(w) => once(DispatcherMessage::Watermark(w)),
        }
    }
}
```